<launch>
    <arg name="ns" default="tracking_ros" />
    <arg name="model_type" default="vit_h" />
    <arg name="device" default="cuda:0" />
    <arg name="mode" default="interactive" />
    <arg name="num_slots" default="-1" />

    <arg name="input_point" default="/$(arg ns)/processed_image/screenpoint" />
    <arg name="input_bbox" default="/$(arg ns)/processed_image/screenrectangle" />
    <arg name="input_image" default="/$(arg ns)/processed_image"/>
    <arg name="prompt_image" default="/$(arg ns)/prompt_image" />
    <arg name="input_segmentation" default="/$(arg ns)/input_segmentation" />

    <node name="sam_node"
            pkg="tracking_ros" type="sam_node.py"
            output="screen" >
        <remap from="~input_image" to="$(arg input_image)" />
        <remap from="~output_image" to="$(arg prompt_image)" />
        <remap from="~input_point" to="$(arg input_point)" />
        <remap from="~input_bbox" to="$(arg input_bbox)" />
        <remap from="~segmentation" to="$(arg input_segmentation)" />
        <rosparam subst_value="true" >
            device: $(arg device)
            model_type: $(arg model_type)
            model_dir: $(find tracking_ros)/checkpoints/
            mode: $(arg mode)
            num_slots: $(arg num_slots)
        </rosparam>
    </node>
</launch>
